package.json

```
@@ -22,6 +22,8 @@
     "lint": "yarn lint:js && yarn lint:styles",
     "lint:js": "eslint . --ext .ts,.tsx,.js",
     "lint:styles": "stylelint src/core/**/styles.ts",
+    "package-validation": "yarn workspace @guardian/source-package-validation",
+    "package-validation:validate": "yarn build:all && yarn package-validation validate",
```

having this as an actual packages in the workspace seems to blur the lines between what are maintenance scripts and what are the things we/contributors are building.

what do you think about something like this?

```suggestion
    "validate-dist": "tsc ./scripts/validate-dist/types.ts && jest ./scripts/validate-dist/package-exports.test.ts",
```

this would also need `build` to have run of course, but doing that manually seems ok?

---

Yeah, I think you're right. When we were recreating `source-shakedown` locally, it seemed to make sense as a package in its own right. But now, it's boiled down to just a couple of scripts and dependencies that come from the workspace root.

---

I agree that this should all live in scripts rather than packages to distinguish what's internal and what is a published package. I'll move that now.

For developing locally, I think having the ease of running `yarn package-validation test:types` and `yarn package-validation test:components` is fairly important. Given that these require custom ts and jest configs, I don't think it's a great experience to have to remember and type `yarn jest ./scripts/validate-dist/package-exports.test.ts --config=./scripts/validate-dist/jest.config.js`. 

Having them in their own package removes the need to explicitly point at the config files and the workspace simplifies the running of the individual commands from the root level without adding multiple commands to the root `package.json` so I'd prefer we either keep it or find a nicer way of achieving the same thing / something similar.

---

> Having them in their own package removes the need to explicitly point at the config files and the workspace simplifies the running of the individual commands from the root level without adding multiple commands to the root package.json so I'd prefer we either keep it or find a nicer way of achieving the same thing / something similar.

this makes a lot of sense 👍 

do you think we still need to set up the yarn alias in this case? maybe something like this?

```suggestion
    "validate-dist": "yarn workspace validate-dist validate",
```

---

Yeah, that looks good. I'll update it to be `validate-dist` everywhere.